cinema/serializers.py

```diff
@@ -68,13 +84,58 @@ class Meta:
             "movie_title",
             "cinema_hall_name",
             "cinema_hall_capacity",
+            "tickets_available"
         )
 
+    @staticmethod
+    def get_tickets_available(obj):
+        return obj.cinema_hall.capacity - obj.tickets.count()
+
 
 class MovieSessionDetailSerializer(MovieSessionSerializer):
     movie = MovieListSerializer(many=False, read_only=True)
     cinema_hall = CinemaHallSerializer(many=False, read_only=True)
+    taken_places = TicketsTakenPlacesSerializer(
+        many=True,
+        read_only=True,
+        source="tickets"
+    )
 
     class Meta:
         model = MovieSession
-        fields = ("id", "show_time", "movie", "cinema_hall")
+        fields = ("id", "show_time", "movie", "cinema_hall", "taken_places")
+
+
+class TicketSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Ticket
+        fields = ("id", "row", "seat", "movie_session")
+
+
+class TicketListSerializer(TicketSerializer):
+    movie_session = MovieSessionListSerializer(many=False, read_only=True)
+
+
+class OrderSerializer(serializers.ModelSerializer):
+    tickets = TicketSerializer(many=True, read_only=False, allow_empty=False)
+
+    class Meta:
+        model = Order
+        fields = ("id", "tickets", "created_at")
+
+    def create(self, validated_data):
+        with transaction.atomic():
+            tickets_data = validated_data.pop("tickets")
+            order = Order.objects.create(**validated_data)
+            for ticket_data in tickets_data:
+                Ticket.objects.create(order=order, **ticket_data)
+            return order
```

Comment:
You can simply use decorator, since whole function is a transaction.


Advice:
Using bulk_create can significantly improve efficiency when creating multiple objects, like Ticket instances. Consider restructuring the code to utilize bulk_create for creating Ticket objects instead of iterating through them in a loop.
Example:
```
ticket_objects = [Ticket(order=order, **ticket_data) for ticket_data in tickets]
Ticket.objects.bulk_create(ticket_objects)
```
using list generator for objects defines better arrangement of memory inside python code, and using bulk_create optimizes memory allocation and is more efficient than a standard for loop for object creation.